src/test.ts

```diff
@@ -1,43 +1,74 @@
+		expect(iobObjStateJSON.object.common).to.have.keys(["role", "name", "type", "read", "write", "category", "desc"]);
+		expect(iobObjStateJSON.object.common.name).to.equal("statejson");
+		expect(iobObjStateJSON.object.common.type).to.equal("string");
+	});
+	it("Build state object with role value", () => {
+		const iobObjStateNumber = iobHelper.buildObject({ namespace: "test.0"} as globalThis.ioBroker.Adapter , {id: "statenumber", name: "statenumber", objectType: "state", role: "value" });
+		iobObjStateNumber.object.common.unit = "°C";
+		//console.log(JSON.stringify(iobObjStateJSON));
+		expect(iobObjStateNumber).to.have.keys(["id", "object"]);
+		expect(iobObjStateNumber.id).to.equal("test.0.statenumber");
+		expect(iobObjStateNumber.object).to.have.keys(["type", "common", "native"]);
+		expect(iobObjStateNumber.object.common).to.have.keys(["role", "name", "type", "read", "write", "category", "desc", "unit"]);
+		expect(iobObjStateNumber.object.common.name).to.equal("statenumber");
+		expect(iobObjStateNumber.object.common.type).to.equal("number");
+		expect(iobObjStateNumber.object.common.unit).to.equal("°C");
+	});
 });
-// Expected error:
-iobObj3.object.common.unit = "m";
+//#endregion
 
-const iobObj4 = buildObject(undefined as any, {
-	id: "statistics.total_distance_meter",
-	name: "total_distance_meter",
-	value: "test",
-	objectType:"state",
-	role: "adapter.messagebox"
+//#region Test validateObject
+describe("Test validateObject", () =>{
+	it("Validate object with successfull result (simple)", () =>{
+		const iobObjs: iobHelper.ObjectWithValue[] = Array(iobHelper.buildObject({ namespace: "test.0"} as globalThis.ioBroker.Adapter , {id: "channel1", name: "channel1", objectType: "template", template: "channel"}));
+		iobObjs.push(iobHelper.buildObject({ namespace: "test.0"} as globalThis.ioBroker.Adapter , {id: "channel1.state1", name: "channel1_state1", objectType: "template", template: "json"}));
+		expect(iobHelper.validateObjectTree.bind(iobHelper.validateObjectTree, iobObjs)).to.not.throw();
+	});
+	it("Validate object with unsuccessfull result (simple)", () =>{
+		const iobObjs: iobHelper.ObjectWithValue[] = Array(iobHelper.buildObject({ namespace: "test.0"} as globalThis.ioBroker.Adapter , {id: "channel1.state1", name: "state1", objectType: "template", template: "json"}));
+		expect(iobHelper.validateObjectTree.bind(iobHelper.validateObjectTree, iobObjs)).to.throw("No superior object declared for test.0.channel1");
+	});
+	it("Validate object with successfull result (complex)", () =>{
+		const iobObjs: iobHelper.ObjectWithValue[] = Array(iobHelper.buildObject({ namespace: "test.0"} as globalThis.ioBroker.Adapter , {id: "foobar", name: "channel1", objectType: "template", template: "channel"}));
+		iobObjs.push(iobHelper.buildObject({ namespace: "test.0"} as globalThis.ioBroker.Adapter , {id: "foobar.state", name: "channel1_state1", objectType: "template", template: "json"}));

```

Comment:
Just define `{ namespace: "test.0"} as globalThis.ioBroker.Adapter` once as a constant somewhere. Also you shouldn't need `globalThis`.